tools/dependency-check.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+failure_exit() {
+  echo >&2 "Could not find ${1}.  Please install that before continuing."
+  exit 1
+}
+
+check_python_headers() {
+  local python_headers_present
+  python_headers_present="$(pkg-config --libs python-3.7)"
+
+  if [ ! "${python_headers_present}" ]; then
+    failure_exit "Python 3.7 headers"
+  fi
+}
+
+check_binary_present() {
+  local binary_exists
+  binary_exists="$(command -v "${1}")"
+  if [ ! "${binary_exists}" ]; then
+    failure_exit "${1}"
+  fi
+}
+
+check_fortran_dependencies() {
+  check_binary_present "gfortran"
+  check_binary_present "f2c"
+}
+
+check_js_dependencies() {
+  check_binary_present "lessc"
+  check_binary_present "uglifyjs"
+}
+
+check_pyyaml() {
+  local pyyaml_import_check
+  pyyaml_import_check="$(python -c 'import yaml')"
```

Comment:
Probably should be `python3`, since that's what the script reading the yaml files will ultimately use.

Reply:
Sounds good.  Changing.